question_four.py

```diff
@@ -0,0 +1,35 @@
+#Create a recursive function which returns the n-th Fibonacci number. [Fibonacci sequence: 1, 1, 2, 3, 5, 8, 13, 21, ...]
+
+def memoize(k): 
+    fast = {}  
+    def partner(l): 
+        if l not in fast: 
+            fast[l] = k(l) 
+        return fast[l] 
+    return partner 
+
+def fibonacci(n): 
+    if (n<=1): 
+        return n  
+    return (fibonacci(n-1) + fibonacci(n-2))  
+fibonacci = memoize(fibonacci)
```

Comment:
The function you have defined is correct. 
But can I rewrite the `fibonacci = memoize(fibonacci)` in another way? Is there any way exist? 
I can provide a hint to you that you have already learned this in the Python, so try to recall it.